core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java

```diff
@@ -41,75 +41,67 @@
+    DetailedEnvironmentResponse edit(@PathParam("workspaceId") Long workspaceId, @PathParam("crn") String environmentCrn,
+            @NotNull EnvironmentEditRequest request);
 
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.LIST, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "listEnvironment")
-    SimpleEnvironmentV4Responses list(@PathParam("workspaceId") Long workspaceId);
+    SimpleEnvironmentResponses list(@PathParam("workspaceId") Long workspaceId);
 
     @PUT
-    @Path("/{name}/attach")
+    @Path("/{crn}/attach")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.ATTACH_RESOURCES, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "attachResourcesToEnvironment")
-    DetailedEnvironmentV4Response attach(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
+    DetailedEnvironmentV4Response attach(@PathParam("workspaceId") Long workspaceId, @PathParam("crn") String environmentCrn,
             @Valid EnvironmentAttachV4Request request);
 
     @PUT
-    @Path("/{name}/detach")
+    @Path("/{crn}/detach")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.DETACH_RESOURCES, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "detachResourcesFromEnvironment")
-    DetailedEnvironmentV4Response detach(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
+    DetailedEnvironmentV4Response detach(@PathParam("workspaceId") Long workspaceId, @PathParam("crn") String environmentCrn,
             @Valid EnvironmentDetachV4Request request);
 
     @PUT
-    @Path("/{name}/change_credential")
-    @Produces(MediaType.APPLICATION_JSON)
-    @ApiOperation(value = EnvironmentOpDescription.CHANGE_CREDENTIAL, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
-            nickname = "changeCredentialInEnvironment")
-    DetailedEnvironmentV4Response changeCredential(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
-        @Valid EnvironmentChangeCredentialV4Request request);
-
-    @PUT
-    @Path("/{name}/register_datalake")
+    @Path("/{crn}/register_datalake")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.REGISTER_EXTERNAL_DATALAKE, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "registerExternalDatalake")
-    DetailedEnvironmentV4Response registerExternalDatalake(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
-        @Valid RegisterDatalakeV4Request request);
+    DetailedEnvironmentV4Response registerExternalDatalake(@PathParam("workspaceId") Long workspaceId, @PathParam("crn") String environmentCrn,
```

Comment:
Shouldn't it be DetailedEnvironmentResponse ?

Reply:
Nope, this is the old Endpoint in cloudbreak which I left here to make it easier to rewire integration tests to use the new env service. If I had deleted this class, most of the integration tests should be deleted too. And it wouldn't be much fun rewriting them.